### ISSUE TITLE
Error: Loaders: Obographs: Unhandled `null`

### ISSUE THREAD

**Issue by joeflack4** [linkml/linkml#1156]
## Overview
I am converting an OWL file to Obographs JSON. I was then using that JSON in OAK, which attempted to do some loading using `linkml-runtime`. `linkml-runtime` encountered an error because `null` was not anticipated.

## Error
Short err: `TypeError: list indices must be integers or slices, not NoneType`

Long err:
```
Traceback (most recent call last):
  ...
    gd: GraphDocument = json_loader.load(inpath, target_class=GraphDocument)
  File "/Users/joeflack4/virtualenvs/tims-ts/lib/python3.10/site-packages/linkml_runtime/loaders/loader_root.py", line 85, in load
    results = self.load_any(*args, **kwargs)
  File "/Users/joeflack4/virtualenvs/tims-ts/lib/python3.10/site-packages/linkml_runtime/loaders/json_loader.py", line 28, in load_any
    return self.load_source(source, loader, target_class,
  File "/Users/joeflack4/virtualenvs/tims-ts/lib/python3.10/site-packages/linkml_runtime/loaders/loader_root.py", line 61, in load_source
    data_as_dict = loader(data, metadata)
  File "/Users/joeflack4/virtualenvs/tims-ts/lib/python3.10/site-packages/linkml_runtime/loaders/json_loader.py", line 22, in loader
    return self.json_clean(data_as_dict)
  File "/Users/joeflack4/virtualenvs/tims-ts/lib/python3.10/site-packages/linkml_runtime/loaders/loader_root.py", line 33, in json_clean
    Loader.json_clean(v)
  File "/Users/joeflack4/virtualenvs/tims-ts/lib/python3.10/site-packages/linkml_runtime/loaders/loader_root.py", line 27, in json_clean
    Loader.json_clean(e)
  File "/Users/joeflack4/virtualenvs/tims-ts/lib/python3.10/site-packages/linkml_runtime/loaders/loader_root.py", line 33, in json_clean
    Loader.json_clean(v)
  File "/Users/joeflack4/virtualenvs/tims-ts/lib/python3.10/site-packages/linkml_runtime/loaders/loader_root.py", line 27, in json_clean
    Loader.json_clean(e)
  File "/Users/joeflack4/virtualenvs/tims-ts/lib/python3.10/site-packages/linkml_runtime/loaders/loader_root.py", line 33, in json_clean
    Loader.json_clean(v)
  File "/Users/joeflack4/virtualenvs/tims-ts/lib/python3.10/site-packages/linkml_runtime/loaders/loader_root.py", line 25, in json_clean
    del(inp[e])
TypeError: list indices must be integers or slices, not NoneType
python-BaseException
```

Screenshot:
<img width="1128" alt="loader err" src="https://user-images.githubusercontent.com/13045020/206626197-7d470957-90cc-4e73-9cba-31daa8435c13.png">

## Possible solutions
I don't intend for the scope of this to necessarily be to handle all `null`/`None` within an Obographs document. But at least for the purposes of parsing `logicalDefinitionAxioms`, which is where this particular issue arose from, when `null`/`None` is encountered, we could:
a. Skip it, and print a warning.
b. Throw a graceful exception.
c. Enable a 'strict' mode for LinkML loaders. If strict, do (b), else (a). I'm fine with defaulting to `strict=True`, so long as the error message prints something about the possibility of disabling it.

If not (c), my preference is for (a), so long as there aren't any unintended side effects I'm unaware of. If we went with (b), even if it was a helpful message explaining exactly what was going on, it would still be a lot more work for the user to deal with, especially if they are not super familiar with OWL / Obographs, the tooling, etc.

## Additional information
The original offending Obographs JSON and the OWL which caused it can be seen here: https://github.com/ontodev/robot/issues/1079

## Related
- https://github.com/ontodev/robot/issues/1079
- https://github.com/geneontology/obographs/issues/89

**Comment by cmungall** [comment]
I don't think there is a bug here. However, it would help to make these issues more reproducible, by providing a link to the exact versionIRI used to download the source file.

Currently the genusId is stated as `required` in the OAK obographs schema, so the linkml runtime framework is doing the correct thing in not instantiating these objects.

To change this behavior let's make an issue in the OAK repo - or you could even do the PR directly, remembering to tag https://github.com/geneontology/obographs/issues/89

**Comment by cmungall** [comment]
OK, I did this as part of this related PR https://github.com/INCATools/ontology-access-kit/pull/414

**Comment by cmungall** [comment]
OK, I see a second issue here, now I have managed to track down what file was used for testing. the mondo json has:

```
"restrictions" : [ null ]
```

this is just wrong. I don't think the jsonloader should try and repair issues like this. It must be fixed upstream. For now you can pre-process e.g. with jq. but this is not a linkml issue

**Comment by cmungall** [comment]
if you do need to make the json conform you can do this:

`jq 'del(..|nulls)' mondo.json > mondo.fixed.json`

**Comment by joeflack4** [comment]
@cmungall Here's the URL, sorry: https://github.com/obophenotype/human-phenotype-ontology/releases/download/v2022-10-05/hp-full.owl

It looks like Mondo had the same issue, as you found.

That is a really cool trick with JQ! I fixed the problem on my end earlier in my Python script: https://github.com/HOT-Ecosystem/tims-ts/blob/owl-on-fhir/bin/convert_owl.py#L118

I agree with you, but I have a slight disagreement. It's definitely a problem upstream on the user end of LinkML, but given that we now see that this is a thing that can happen in the wild, and I'm not sure how long that will persist for (you might have a better idea), I feel like exception handling is prudent. In my OP I recommended one of:
a. Skip it, and print a warning.
b. Throw a graceful exception.
c. Enable a 'strict' mode for LinkML loaders. If strict, do (b), else (a). I'm fine with defaulting to strict=True, so long as the error message prints something about the possibility of disabling it.

So at the least I would think (b) would be nice, and I can add this for you if you feel it's worth adding. I'm not sure what it's best to display to the user though. It looked to me like robot couldn't understand an `owl:Restriction` inside of another `owl:Restriction`, and that's what resulted in the `null`. Though it did only happen on `genusId` axioms as well. I could include that information, or even use the report about the GitHub page (which has JSON snippets), like a link to an error page. Well, maybe not the link, but I could include a small JSON snippet in the error message.

I personally would not mind (c) kinda as a leaning preference but don't feel strongly about it.
